File: src/backend/studio-reservation/src/main/java/org/taonaw/studio_reservation/domain/model/openingHourSetting/OpeningHour.java
package org.taonaw.studio_reservation.domain.model.openingHourSetting;

import lombok.EqualsAndHashCode;
import lombok.Getter;
import org.taonaw.studio_reservation.domain.model.shared.TimeRange;
import org.taonaw.studio_reservation.domain.shared.Assertion;

import java.time.LocalTime;

@Getter
@EqualsAndHashCode(callSuper = true)
public class OpeningHour extends TimeRange {
    public OpeningHour(LocalTime startTime, LocalTime endTime) {
        super(startTime, endTime);
        Assertion.required(startTime);
        Assertion.required(endTime);

        if (hasStartTimeSeconds()) {
            throw new IllegalArgumentException("開始時間に秒の指定はできません。");
        }
        if (hasEndTimeSeconds()) {
            throw new IllegalArgumentException("終了時間に秒の指定はできません。");
        }
    }

    public boolean isAllDay() {
        return getStartTime().equals(getEndTime());
    }
}

File: src/backend/facility/src/main/kotlin/org/taonaw/facility/domain/model/tenant/OpeningHour.kt
package org.taonaw.facility.domain.model.tenant

import java.time.LocalTime

data class OpeningHour(
        val start: LocalTime,
        val end: LocalTime) {

    init {
        require(start.second == 0 && start.nano == 0) { "開始日時に秒の指定はできません。" }
        require(end.second == 0 && end.nano == 0) { "終了日時に秒の指定はできません。" }
    }
}